csrc/fusion.cpp

```diff
@@ -683,6 +683,7 @@ std::vector<Val*> Fusion::getTerminatingOutputs() const {
   auto is_reachable_to_output = [](Val* val) {
```

Comment:
I thought these are already removed from `uses()`.

Reply:
hmm. That's a good point. I'll refactor this.

Frankly speaking, the changes here isn't really needed any more, since calling `uses()` would already reset uses and go through traverseBetween, that would throw on a cyclic fusion. :laughing: